tests/core/test_entity.py

```diff
@@ -738,3 +725,458 @@ def test_first_with_cache(self):
+
+    def test_multiple_excludes(self):
+        """Test query construction with multiple exclude criteria"""
+        # Filter by the Owner
+        q1 = Dog.query.exclude(owner='John')
+        q2 = q1.exclude(age=3)
+
+        _, decon_args, decon_kwargs = q2._criteria.deconstruct()
+
+        assert str(decon_args) == ("(<Q: (NOT (AND: ('owner', 'John')))>, "
+                                   "<Q: (NOT (AND: ('age', 3)))>)")
+        assert decon_kwargs == {}
+        assert q2._criteria.connector == Q.AND
+        assert q2._criteria.negated is False
+        assert Q(*decon_args, **decon_kwargs) == q2._criteria
+
+    def test_multiple_criteria_in_filter(self):
+        """Test query construction with multiple filter in filter"""
+        # Filter by the Owner
+        q1 = Dog.query.filter(owner='John', age=3)
+        _, decon_args, decon_kwargs = q1._criteria.deconstruct()
+
+        assert decon_args == (('age', 3), ('owner', 'John'))
+        assert decon_kwargs == {}
+        assert q1._criteria.connector == Q.AND
+        assert q1._criteria.negated is False
+        assert Q(*decon_args, **decon_kwargs) == q1._criteria
+
+    def test_multiple_criteria_in_exclude(self):
+        """Test query construction with multiple filter in exclude"""
+        # Filter by the Owner
+        q1 = Dog.query.exclude(owner='John', age=3)
+
+        _, decon_args, decon_kwargs = q1._criteria.deconstruct()
+
+        assert decon_args == (('age', 3), ('owner', 'John'))
+        assert decon_kwargs == {'_negated': True}
+        assert q1._criteria.connector == Q.AND
+        assert q1._criteria.negated is True
+        assert Q(*decon_args, **decon_kwargs) == q1._criteria
+
+    def test_combined_filter_and_exclude(self):
+        """Test query construction with combined filter/exclude with filter coming first"""
+        # Filter by the Owner
+        q1 = Dog.query.filter(owner='John').exclude(age=3)
+
+        _, decon_args, decon_kwargs = q1._criteria.deconstruct()
+
+        assert str(decon_args) == "(('owner', 'John'), <Q: (NOT (AND: ('age', 3)))>)"
+        assert decon_kwargs == {}
+        assert q1._criteria.connector == Q.AND
+        assert q1._criteria.negated is False
+        assert Q(*decon_args, **decon_kwargs) == q1._criteria
+
+    def test_combined_exclude_and_filter(self):
+        """Test query construction with combined filter/exclude with exclude coming first"""
+        # Filter by the Owner
+        q1 = Dog.query.exclude(age=3).filter(owner='John')
+
+        _, decon_args, decon_kwargs = q1._criteria.deconstruct()
+
+        assert str(decon_args) == "(<Q: (NOT (AND: ('age', 3)))>, ('owner', 'John'))"
+        assert decon_kwargs == {}
+        assert q1._criteria.connector == Q.AND
+        assert q1._criteria.negated is False
+        assert Q(*decon_args, **decon_kwargs) == q1._criteria
+
+    def test_filter_with_single_q_object(self):
+        """Test query construction with negation"""
```

Comment:
Need to fix the doc string for these test cases...

Reply:
Fixed for all cases in this class.